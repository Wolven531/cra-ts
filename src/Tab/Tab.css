@import '../vars.css';

.tabs ul li {
	border: 1px solid var(--halloween-orange);
	border-radius: 5px;
	box-shadow: 0 0 5px 5px var(--halloween-black-faded);
	box-sizing: border-box;
	padding: var(--spacing);
	color: var(--halloween-white);
	cursor: pointer;
	min-width: 80px;
	text-align: center;
	text-shadow: 1px 1px var(--halloween-black-faded);
	text-transform: capitalize;
}

.tabs ul li.selected {
	color: var(--halloween-yellow);
	font-weight: bold;
}
