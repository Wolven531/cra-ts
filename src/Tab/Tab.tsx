import React, { FC } from 'react'
import './Tab.css'

export interface TabProps {
	onClick: (tabNum: number) => void
	selected: boolean
	tabNum: number
}

export const Tab: FC<TabProps> = ({ onClick, selected, tabNum }) => {
	return (
		<li
			className={selected ? 'selected' : ''}
			onClick={() => {
				onClick(tabNum)
			}}
		>
			tab {tabNum}
		</li>
	)
}
