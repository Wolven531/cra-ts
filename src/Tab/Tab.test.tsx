import React from 'react'
import { render, RenderResult, waitFor } from '@testing-library/react'
import { Tab } from './Tab'
import { MemoryRouter, Route } from 'react-router-dom'

describe('Tab', () => {
	const fakeTabNum = 3
	let comp: RenderResult
	let mockOnClick: jest.Mock

	beforeEach(() => {
		mockOnClick = jest.fn()

		comp = render(
			<MemoryRouter initialEntries={['/tabs']} initialIndex={0}>
				<Route path={['/tabs/:tabId', '/tabs']}>
					<Tab onClick={mockOnClick} selected tabNum={fakeTabNum} />
				</Route>
			</MemoryRouter>
		)
	})

	it('matches loading snapshot', () => {
		expect(comp.asFragment()).toMatchSnapshot()
	})

	describe('when loaded and selected', () => {
		beforeEach(async () => {
			await waitFor(() => comp.getByTestId('tab'))
		})

		it('matches snapshot', () => {
			expect(comp.asFragment()).toMatchSnapshot()
		})
	})
})

describe('Tab when loaded and not selected', () => {
	let comp: RenderResult
	let mockOnClick: jest.Mock

	beforeEach(async () => {
		mockOnClick = jest.fn()

		comp = render(
			<MemoryRouter initialEntries={['/tabs/1']} initialIndex={0}>
				<Route path={['/tabs/:tabId', '/tabs']}>
					<Tab onClick={mockOnClick} selected={false} tabNum={2} />
				</Route>
			</MemoryRouter>
		)

		await waitFor(() => comp.getByTestId('tab'))
	})

	it('matches snapshot', () => {
		expect(comp.asFragment()).toMatchSnapshot()
	})
})
