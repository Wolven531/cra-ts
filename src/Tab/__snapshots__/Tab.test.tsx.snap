// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab matches loading snapshot 1`] = `
<DocumentFragment>
  <li
    class="selected"
    data-testid="tab"
  >
    tab 3
  </li>
</DocumentFragment>
`;

exports[`Tab when loaded and not selected matches snapshot 1`] = `
<DocumentFragment>
  <li
    class=""
    data-testid="tab"
  >
    tab 2
  </li>
</DocumentFragment>
`;

exports[`Tab when loaded and selected matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="selected"
    data-testid="tab"
  >
    tab 3
  </li>
</DocumentFragment>
`;
